sdk/schemaregistry/schema-registry-json/test/public/errors.spec.ts

```diff
@@ -289,627 +137,105 @@ describe("Error scenarios", function () {
+    it("schema without an ID", async function () {
       await assertError(
         serializer.serialize(
           null,
           JSON.stringify({
-            type: "record",
-            name: "foo",
+            $schema: "https://json-schema.org/draft/2020-12/schema",
+            title: "Student",
+            description: "A student in the class",
+            type: "string",
           })
         ),
         {
-          causeMessage: /non-array record fields: undefined/,
+          message: /Schema must have an ID/,
         }
       );
     });
-  });
-  describe("Unserialized value validation", function () {
-    it("schema is still registered if serialization fails", async function (this: Context) {
-      /**
-       * This test checks for service calls using the onResponse callback but
-       * onResponse is not implemented in the mocked registry because it will
-       * add very little value so the test is skipped in playback mode.
-       */
+    it("schema with invalid type", async function () {
       if (!isLiveMode()) {
         this.skip();
       }
-      let ran = false;
-      const unusedRegistry = createTestRegistry({
-        registerSchemaOptions: {
-          onResponse: () => {
-            ran = true;
-          },
-        },
-        recorder,
-      });
-      const customSerializer = await createTestSerializer({
-        registry: unusedRegistry,
-        serializerOptions: {
-          autoRegisterSchemas: true,
-          groupName: testGroup,
-        },
-        recorder,
-      });
-      await assertError(
-        customSerializer.serialize(
-          {
-            field: 1,
-          },
-          JSON.stringify({
-            type: "record",
-            name: "User",
-            namespace: "validation",
-            fields: [
-              {
-                name: "field",
-                type: "null",
-              },
-            ],
-          })
-        ),
-        {
-          causeMessage: /invalid "null": 1/,
-        }
-      );
-      assert.isTrue(ran, `Expected a service call to register the schema but non was sent!`);
-    });
-    it("null", async function () {
       await assertError(
         serializer.serialize(
           {
-            field: 1,
+            name: "Anne",
           },
           JSON.stringify({
-            type: "record",
-            name: "User",
-            namespace: "validation",
-            fields: [
-              {
-                name: "field",
-                type: "null",
-              },
-            ],
+            $schema: "https://json-schema.org/draft/2020-12/schema",
+            $id: "student",
+            title: "Student",
+            description: "A student in the class",
+            type: "none",
           })
         ),
         {
-          causeMessage: /invalid "null": 1/,
+          message: /Invalid JSON schema type: none/,
         }
       );
     });
-    it("boolean", async function () {
```

Comment:
Should we add back all those type tests and use the validate callback?

Reply:
Perhaps we can move them to the validation test suite

Reply:
The tests are added to the validation test suite